pkg/registry/pod/etcd/etcd.go

```diff
@@ -110,11 +121,11 @@ func NewStorage(s storage.Interface, useCacher bool, k client.ConnectionInfoGett
 		Pod:         &REST{store, proxyTransport},
 		Binding:     &BindingREST{store: store},
 		Status:      &StatusREST{store: &statusStore},
-		Log:         &podrest.LogREST{Store: store, KubeletConn: k},
+		Log:         &podrest.LogREST{Store: store, HostLocator: hostLocator, KubeletConn: k},
 		Proxy:       &ProxyREST{store: store, proxyTransport: proxyTransport},
-		Exec:        &ExecREST{store: store, kubeletConn: k},
-		Attach:      &AttachREST{store: store, kubeletConn: k},
-		PortForward: &PortForwardREST{store: store, kubeletConn: k},
+		Exec:        &ExecREST{store: store, hostLocator: hostLocator, kubeletConn: k},
```

Comment:
Ideally it should, but it would require large refactoring. As I mentioned earlier, when we move information about KubeletPort (and possibly other stuff in the future, like tokens or sth.) to the etcd we need query it before creating a connection. In this solution hostLocator is passed to kubeletConn, so that it's able to read a Port from etcd, which makes client simple (etcd is abstracted away). Other option, which is adding etcd handling to the client is much worse.